[BUG]: c4800 warning when compiling with MSVC and python 3.10

### Required prerequisites

- [X] Make sure you've read the [documentation](https://pybind11.readthedocs.io). Your issue may be addressed there.
- [X] Search the [issue tracker](https://github.com/pybind/pybind11/issues) and [Discussions](https:/pybind/pybind11/discussions) to verify that this hasn't already been reported. +1 or comment there if it has.
- [X] Consider asking first in the [Gitter chat room](https://gitter.im/pybind/Lobby) or in a [Discussion](https:/pybind/pybind11/discussions/new).

### Problem description

Python 3.9 to Python 3.10 changed this code:
```
#define PyObject_TypeCheck(ob, tp) \
    (Py_IS_TYPE(ob, tp) || PyType_IsSubtype(Py_TYPE(ob), (tp)))
```

to [this code](https://github.com/python/cpython/blob/3.10/Include/object.h):
```
static inline int _PyObject_TypeCheck(PyObject *ob, PyTypeObject *type) {
    return Py_IS_TYPE(ob, type) || PyType_IsSubtype(Py_TYPE(ob), type);
}
#define PyObject_TypeCheck(ob, type) _PyObject_TypeCheck(_PyObject_CAST(ob), type)
```

That change gives two c4800 warnings [here](https://github.com/pybind/pybind11/blob/master/include/pybind11/numpy.h#L173):
```
    bool PyArray_Check_(PyObject *obj) const {
        return (bool) PyObject_TypeCheck(obj, PyArray_Type_);
    }
    bool PyArrayDescr_Check_(PyObject *obj) const {
        return (bool) PyObject_TypeCheck(obj, PyArrayDescr_Type_);
    }
```

`pybind11/numpy.h(173): warning C4800: 'int': forcing value to bool 'true' or 'false' (performance warning)`
`pybind11/numpy.h(176): warning C4800: 'int': forcing value to bool 'true' or 'false' (performance warning)`


Consider changing to something like this given [you've done similar fixes](https://github.com/pybind/pybind11/commit/e93d94594b063682a70ead0b7feb8c1329ae0717):
```
    bool PyArray_Check_(PyObject *obj) const {
        return PyObject_TypeCheck(obj, PyArray_Type_) != 0;
    }
    bool PyArrayDescr_Check_(PyObject *obj) const {
        return PyObject_TypeCheck(obj, PyArrayDescr_Type_) != 0;
    }
```


P.S. I'm sorry this is not a PR. But I don't wanna set up everything for such a small change

### Reproducible example code

_No response_

--- a/include/pybind11/numpy.h
+++ b/include/pybind11/numpy.h
@@ -170,10 +170,10 @@ struct npy_api {
     }
 
     bool PyArray_Check_(PyObject *obj) const {
-        return (bool) PyObject_TypeCheck(obj, PyArray_Type_);
+        return PyObject_TypeCheck(obj, PyArray_Type_) != 0;
     }
     bool PyArrayDescr_Check_(PyObject *obj) const {
-        return (bool) PyObject_TypeCheck(obj, PyArrayDescr_Type_);
+        return PyObject_TypeCheck(obj, PyArrayDescr_Type_) != 0;
     }
 
     unsigned int (*PyArray_GetNDArrayCFeatureVersion_)();

